### ISSUE TITLE
react-client has no way to release the microphone

### ISSUE THREAD

**Issue by dsernst** [speechly/speechly#304]
### Describe the bug
As described in https://github.com/speechly/speechly/issues/226, there's no way to disconnect the microphone after calling @speechly/react-client's `stop()`.

This is infecting every other browser window with a Microphone listening icon, very unpleasant. On iOS it creates a persistent orange microphone icon in the top left corner, even after switching out of the browser.

### To Reproduce
1. Call `attachMicrophone()` and `start()` from the [React Getting Started tutorial](https://docs.speechly.com/guides/react-client#capture-microphone-audio)
2. Call `stop()`

### Expected behaviour
Expect the browser to stop reporting that it's listening.

Passing in a flag like `stop({ disconnect: true })` is fine, it doesn't need to be the default, just need some option.

### Screenshots
Safari at least is infecting all windows with this (incl the one I'm writing this issue from! 😄), with no way to stop it other than refreshing or closing the page that called Speechly.

<img width="380" alt="image" src="https://user-images.githubusercontent.com/6340841/236935443-fe344473-b931-4cfe-8a08-75cc4c97343e.png">

### Environment

- Platform: `@speechly/react-client` v2.2.3
- Browser: Safari 16.3

**Comment by dsernst** [comment]
This [`patch-package`](https://www.npmjs.com/package/patch-package) tempfix worked for me:

```diff
diff --git a/node_modules/@speechly/react-client/dist/context.js b/node_modules/@speechly/react-client/dist/context.js
index 8716670..ae1bc86 100644
--- a/node_modules/@speechly/react-client/dist/context.js
+++ b/node_modules/@speechly/react-client/dist/context.js
@@ -179,10 +179,12 @@ var SpeechProvider = /** @class */ (function (_super) {
             var client;
             return __generator(this, function (_a) {
                 client = this.state.client;
+                var microphone = this.state.microphone
+                if (microphone) microphone.close()
                 this.setState({ listening: false });
                 if (client == null) {
                     throw Error('No Speechly client (are you calling stop in non-browser environment)');
                 }
                 return [2 /*return*/, client.stop()];
             });
         }); };
```

**Comment by thisismatu** [comment]
Hi 👋 

The `start()` and `stop()` methods are used for controlling audio processing. It's intentional that `stop()` does not disconnect the microphone.

However, `microphone` gives you low-level access to underlying BrowserMicrophone and it has a `close()` method that does what you are looking for. https://github.com/speechly/speechly/blob/main/libraries/browser-client/docs/classes/microphone.BrowserMicrophone.md#close

I just tested this and using [browser-client](https://www.npmjs.com/package/@speechly/browser-client) it does work, but [react-client](https://www.npmjs.com/package/@speechly/react-client) for som reason fails silently. 

So thanks for pointing out this bug, we'll look into it.

**Comment by thisismatu** [comment]
The issue should now have been resolved in the `2.2.4` release!

Here's some example code showing how to disconnect the microphone.

```jsx
const { microphone, listening, start, stop, attachMicrophone } = useSpeechContext();

const handleStartStop = async () => {
  if (listening) {
    await stop();
  } else {
    await attachMicrophone();
    await start();
  }
};

const handleDisconnect = async () => {
  if (microphone) {
    await stop();
    await microphone.close();
  }
};

return (
  <div className="App">
    <button onClick={handleStartStop}>
      {listening ? 'Stop' : 'Start'} microphone
    </button>
    <button onClick={handleDisconnect}>
      Disconnect microphone
    </button>
  </div>
);
```

Closing this issue as it was resolved.

**Comment by dsernst** [comment]
great. thanks!
